packages/dds/tree/src/feature-libraries/sequence-field/compose.ts

```diff
@@ -62,62 +62,30 @@ function composeMarkLists<TNodeChange>(
     composeChild: NodeChangeComposer<TNodeChange>,
 ): MarkList<TNodeChange> {
     const factory = new MarkListFactory<TNodeChange>();
-    const baseIter = new StackyIterator(baseMarkList);
-    const newIter = new StackyIterator(newMarkList);
-    for (let newMark of newIter) {
-        let baseMark: Mark<TNodeChange> | undefined = baseIter.pop();
-        if (baseMark === undefined) {
-            // We have reached a region of the field that the base change does not affect.
-            // We therefore adopt the new mark as is.
+    const queue = new ComposeQueue(baseMarkList, newRev, newMarkList);
+    while (!queue.isEmpty()) {
+        const { baseMark, newMark, areInverses } = queue.pop();
+        if (areInverses) {
+            continue;
+        }
+        if (newMark === undefined) {
+            assert(baseMark !== undefined, "Non-empty queue should not return two undefined marks");
+            factory.push(baseMark);
+        } else if (baseMark === undefined) {
+            assert(newMark !== undefined, "Non-empty queue should not return two undefined marks");
```

Comment:
Does the TS compiler not infer that one?

Reply:
It does. I think I wrote the assert before adding the first if-block.